service/history/api/startworkflow/api.go

```diff
@@ -431,6 +438,35 @@ func (s *Starter) resolveDuplicateWorkflowID(
 		return nil, StartNew, nil
 	}
 
+	killOnce.Do(func() {
+		err = api.GetAndUpdateWorkflowWithNew(
+			ctx,
+			nil,
+			definition.NewWorkflowKey(
+				s.namespace.ID().String(),
+				workflowID,
+				currentWorkflowConditionFailed.RunID,
+			),
+			func(workflowLease api.WorkflowLease) (*api.UpdateWorkflowAction, error) {
+				mutableState := workflowLease.GetMutableState()
+				return api.UpdateWorkflowTerminate, workflow.TerminateWorkflow(
+					mutableState,
+					"TESTING",
+					nil,
+					"TESTING",
+					false,
+					nil,
+				)
+			},
+			nil,
+			s.shardContext,
+			s.workflowConsistencyChecker,
+		)
+		if err != nil {
+			panic(err)
+		}
+	})
```

Comment:
Obviously, I had to remove this again, but it demonstrates that the change works.

Reply:
<img width="709" alt="Screenshot 2025-01-17 at 10 06 02 AM" src="https://github.com/user-attachments/assets/70caa354-81c6-47a3-9cc0-60bfe1883194" />
<img width="1010" alt="Screenshot 2025-01-17 at 10 05 55 AM" src="https://github.com/user-attachments/assets/6f656c7d-4962-4736-8d6f-b8ffc499d934" />